cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -539,13 +549,21 @@ void aggregate_reader_metadata::column_info_for_row_group(row_group_info& rg_inf
 }
 
 aggregate_reader_metadata::aggregate_reader_metadata(
-  host_span<std::unique_ptr<datasource> const> sources, bool use_arrow_schema)
+  host_span<std::unique_ptr<datasource> const> sources,
+  bool use_arrow_schema,
+  bool has_cols_from_mismatched_srcs)
   : per_file_metadata(metadatas_from_sources(sources)),
     keyval_maps(collect_keyval_metadata()),
+    schema_idx_maps(init_schema_idx_maps(has_cols_from_mismatched_srcs)),
     num_rows(calc_num_rows()),
     num_row_groups(calc_num_row_groups())
 {
-  if (per_file_metadata.size() > 0) {
+  // Ensure we have at least one datasource to work with.
+  CUDF_EXPECTS(per_file_metadata.size() > 0, "No valid Parquet datasource to be read");
```

Comment:
Do we really need to check it here? Will we ever encounter this?

Reply:
Only when `sources` is empty. 
I think this is possible, since `datasource::create` allows an empty vector of arguments form which we create individual sources.

Reply:
I do wonder if we want to skip steps instead of throwing. I could not find this is tests, but maybe we're expected to return an empty table when reading zero sources.

Reply:
Yes, that's what I wasn't sure about either. I will investigate this a bit more